tests/conftest.py

```diff
@@ -277,6 +280,27 @@ def _fail_for_invalid_template_variable_improved(_fail_for_invalid_template_vari
         )
 
 
+@pytest.fixture(autouse=True, scope="function")
+def unknown_variable_template_error(monkeypatch, request):
+    marker = request.keywords.get("ignore_unknown_variable_template_error", None)
+    if os.environ.get(INVALID_TEMPLATE_VARS_ENV, "false") == "true" and marker is None:
+        origin_resolve = base_template.FilterExpression.resolve
+
+        def stricter_resolve(self, context, ignore_failures=False):
+            if (
+                self.is_var
+                and self.var.lookups is not None
+                and self.var.lookups[0] not in context
+                # debug can be injected by django.template.context_processors.debug
+                # user can be injected by django.contrib.auth.context_processors.auth but should be fixed
```

Comment:
but should be fixed = todo : do not use `user` from `django.contrib.auth.context_processors.auth` and prefer `request.user` ?